### PR TITLE
Edits to Boxplot- adding swap checkbox and removing group parameter

### LINKED ISSUE #4211: Boxplots (and others) by more than factor columns?
The command in ggplot as follows

ggplot(nass.corn, aes(year, acres)) + geom_boxplot(aes(group=year))

Seems to indicate that ggplot can cope with boxplots by year, where year is numeric.  That will allow much more flexible x-axis labelling.  

I wonder how this can be incorporated into our dialogues - at least for boxplots, but perhaps there is a more general option here?  It may need some thought and discussion.

### PR DESCRIPTION
**Pull request by maxwellfundi**
Fixes #4211 

@africanmathsinitiative/developers  this is ready for review :-)

### CODE REVIEW

**Comment by shadrackkibet** [comment]
Reviewing and testing this.

File: instat/dlgBoxPlot.vb [hunk 1]
```diff
@@ -14,6 +14,7 @@
 ' You should have received a copy of the GNU General Public License 
 ' along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+Imports instat
```
Comment (shadrackkibet) [inline instat/dlgBoxPlot.vb]: I think we don't need this.

Comment (maxwellfundi) [reply instat/dlgBoxPlot.vb]: Yes you are right :-), been there for ages i guess

**Comment by shadrackkibet** [comment]
I suggest the swap checkbox to be visible but disabled rather than being hidden completely.Otherwise this is working fine.

@dannyparsons this is ready for you to review.

**Comment by maxwellfundi** [comment]
Thanks @shadrackkibet .

File: instat/dlgBoxPlot.vb [hunk 2]
```diff
@@ -149,9 +154,9 @@ Public Class dlgBoxplot
         clsRgeomPlotFunction.SetRCommand("geom_boxplot")
         clsRgeomPlotFunction.AddParameter("varwidth", "FALSE")
 
-        clsLocalRaesFunction.SetPackageName("ggplot2")
-        clsLocalRaesFunction.SetRCommand("aes")
-        clsRgeomPlotFunction.AddParameter("boxaes", clsRFunctionParameter:=clsLocalRaesFunction, iPosition:=1, bIncludeArgumentName:=False)
+        'clsLocalRaesFunction.SetPackageName("ggplot2")
```
Comment (Patowhiz) [inline instat/dlgBoxPlot.vb]: @maxwellfundi . This is probably done intentionally or not part of the issue you are solving. But just to understand how the RInstant code is implemented differently in various dialogs I'm forced to ask about this. Is this bit of  
         ```
         Dim clsCoordFlipFunc As New RFunction
         Dim clsCoordFlipParam As New RParameter
```
in the `InitialiseDialog()` supposed to be in setDefaults() ?
And the `clsBaseOperator `and the others aren't they supposed to be re-initialised in the setDefaults to ensure a fresh start on click Reset?

File: instat/dlgBoxPlot.vb [hunk 3]
```diff
@@ -116,8 +116,8 @@ Public Class dlgBoxplot
         ucrSaveBoxplot.SetAssignToIfUncheckedValue("last_graph")
 
 
-        'this control exists but hidden for now
-        ' ucrChkSwapParameters.SetText("swap Parameters")
+        'this control exists but diabled for now
```
Comment (shadrackkibet) [inline instat/dlgBoxPlot.vb]: s?

File: instat/dlgBoxPlot.vb (hunk 2 above)
Comment (maxwellfundi) [reply instat/dlgBoxPlot.vb]: So if you follow keenly, we have a parameter (coord_flip())  added to the base operator by a checkbox. So whats happening here is defining the function , which then is sort of set as the parameter of the checkbox which passes it to the baseoperator. No more parameters are added to  the coord_flip() and so its okay to do it this way

  'clsCoordFlipFunc.SetPackageName("ggplot2")
        clsCoordFlipFunc.SetRCommand("coord_flip")
        clsCoordFlipParam.SetArgumentName("coord_flip")
        clsCoordFlipParam.SetArgument(clsCoordFlipFunc)
        ucrChkHorizontalBoxplot.SetText("Horizontal Plot")
        ucrChkHorizontalBoxplot.SetParameter(clsCoordFlipParam, bNewChangeParameterValue:=False, bNewAddRemoveParameter:=True)'

Comment (Patowhiz) [reply instat/dlgBoxPlot.vb]: Okay

**Comment by maxwellfundi** [comment]
@dannyparsons  This is ready for you. it works properly

**Pull request merged**